lib/set.gi

```diff
@@ -222,3 +222,50 @@ InstallMethod( SubtractSet,
       RemoveSet( set, obj );
     od;
     end );
+
+
+#############################################################################
+##
+InstallMethod(SetAsDomain,
+"for a set & collection",
+[IsSet and IsCollection],
+function(set)
+    local copy, type, domain;
+    copy := Immutable(set);
+    type := NewType(FamilyObj(copy),
+                    IsSetAsDomain and IsAttributeStoringRep);
+    return ObjectifyWithAttributes(rec(), type,
+                                   GeneratorsOfDomain, copy,
+                                   AsList, copy,
+                                   AsSet, copy);
+end);
+
+# Provide family of objects `fam`. Then the resulting domain lives in the same
+# family as a domain created from a non-empty set of objects of `fam`.
+InstallOtherMethod(SetAsDomain,
+"for a family and an empty list",
+[IsFamily, IsEmpty and IsList],
+function(family, set)
+    local copy, type;
+    copy := Immutable([]);
+    type := NewType(CollectionsFamily(family),
+                    IsSetAsDomain and IsAttributeStoringRep);
+    return ObjectifyWithAttributes(rec(), type,
+                                   GeneratorsOfDomain, copy,
+                                   AsList, copy,
+                                   AsSet, copy);
+end);
+
+InstallMethod(\in,
+"for an IsSetAsDomain",
+[IsObject, IsSetAsDomain],
```

Comment:
But `IsSetAsDomain` is merely a synonym for `IsDomain`; so now you really have installed a generic `\in` method for arbitrary domains, which tries to use `AsList`. That makes me a bit nervous, though I can't put my finger on it right now.